pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -279,6 +279,15 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(ctx context.Context, request r
 
 	err = cr.CheckNSetDefaults(r.serverVersion.Platform, log)
 	if err != nil {
+		// If the user created a cluster with finalizers and wrong options, it would be impossible to delete a cluster.
+		// We need to run checkFinalizers to delete finalizers
+		if cr.DeletionTimestamp != nil {
+			_, err := r.checkFinalizers(ctx, cr)
+			if err != nil {
+				log.Error(err, "failed to check finalizers")
+			}
+		}
```

Comment:
why don't we run the finalizer check few lines below before `CheckNSetDefaults`?

Reply:
The`checkFinalizers` below allows us to continue reconciling after the call. If `CheckNSetDefaults` fails, it means that we can get nil pointer reference panics after the `checkFinalizers` call. 

But I noticed that it's possible to get these panics even in the `checkFinalizers` method, so I updated the code to just delete finalizers instead of calling `checkFinalizers`